pkg/scheduler/scheduler_test.go

```diff
@@ -430,17 +430,17 @@ func TestSchedule(t *testing.T) {
 			},
 			wantScheduled: []string{"eng-beta/new"},
 		},
-		"cannot borrow if cohort was assigned": {
+		"cannot borrow if cohort was assigned and would result in overadmission": {
```

Comment:
Can you add two more test cases where two workloads are trying to preempt: one test case where they target the same flavor, and one where they target a different flavor.